timestep.c

```diff
@@ -721,8 +768,50 @@ void rebuild_activelist(void)
 }
 
 
+/*! This function finds the next synchronization point of the system
+ * (i.e. the earliest point of time any of the particles needs a force
+ * computation), and drifts the system to this point of time.  If the
+ * system drifts over the desired time of a snapshot file, the
+ * function will drift to this moment, generate an output, and then
+ * resume the drift.
+ */
+unsigned int find_next_kick(unsigned int Ti_Current)
+{
+    int n, ti_next_kick_global;
+    int ti_next_kick = TIMEBASE;
+    /*This repopulates all timebins on the first timestep*/
+    if(TimeBinCount[0])
+        ti_next_kick = Ti_Current;
+
+    /*Extract large (snapshot number) part of Ti_Current*/
+    int snap = Ti_Current & ~(TIMEBASE-1);
+    /*Zero all upper bits of Ti_Current*/
+    Ti_Current &= TIMEBASE-1;
+
+    /* find the next kick time:
+     * this finds the smallest active bin whose bit is not already set in Ti_Current. */
+    for(n = 1; n < TIMEBINS; n++)
```

Comment:
I thought about it and indeed all the bit work is unnecessary. What is committed in 3fb596f should be equivalent to what was there before.